gsad/src/gsad_gmp.c

```diff
@@ -1255,11 +1256,27 @@ get_many (gvm_connection_t *connection, const char *type,
             }
           else if (str_equal (type, "result"))
             {
+              gchar *report_id_filter;
+              and_report_id = params_value (params, "_report_id");
+
+              if (and_report_id == NULL || str_equal (and_report_id, ""))
+                {
+                  report_id_filter =
+                    g_strdup_printf ("_and_report_id=%s", and_report_id);
```

Comment:
I think this assignment is in the block for the wrong condition as it's writing the keyword if the parameter is missing or empty.
It looks like it should be in the `else` block instead while the one from there should be here.

Reply:
You are completely right. I've mixed up both branches.